src/Traces.Web/Services/Apaleo/ApaleoUserClaimValidatorService.cs

```diff
@@ -11,32 +14,57 @@ public class ApaleoUserClaimValidatorService : IApaleoUserClaimValidatorService
     {
         private readonly NavigationManager _navigationManager;
         private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly ILogger _logger;
 
         public ApaleoUserClaimValidatorService(
             NavigationManager navigationManager,
-            IHttpContextAccessor httpContextAccessor)
+            IHttpContextAccessor httpContextAccessor,
+            ILogger<ApaleoUserClaimValidatorService> logger)
         {
-            _navigationManager = navigationManager;
-            _httpContextAccessor = httpContextAccessor;
+            _navigationManager = Check.NotNull(navigationManager, nameof(navigationManager));
+            _httpContextAccessor = Check.NotNull(httpContextAccessor, nameof(httpContextAccessor));
+            _logger = Check.NotNull(logger, nameof(logger));
         }
 
         public void AssertClaim(string queryParameter, string claimType)
         {
-            var extractedQueryParameter = UrlQueryParameterExtractor.ExtractQueryParameterFromManager(_navigationManager, queryParameter);
+            _logger.LogInformation("================== Tracing ==================");
+            _logger.LogInformation($"AssertClaim called with parameters: <{queryParameter}>, <{claimType}>");
 
+            var extractedQueryParameter = UrlQueryParameterExtractor.ExtractQueryParameterFromManager(_navigationManager, queryParameter);
             if (string.IsNullOrWhiteSpace(extractedQueryParameter))
             {
+                _logger.LogInformation($"RETURN: extractedQueryParameter is null, empty or whitespace: <{extractedQueryParameter}>");
                 return;
             }
 
+            _logger.LogInformation($"extractedQueryParameter is <{extractedQueryParameter}>");
+
             var user = _httpContextAccessor.HttpContext.User;
-            var claimValue = user.Claims.FirstOrDefault(x => x.Type == claimType)?.Value;
+            if (user == null || !user.Identity.IsAuthenticated)
+            {
+                _logger.LogInformation($"RETURN: user is null or unauthenticated");
+                return;
+            }
 
-            if (!extractedQueryParameter.Equals(claimValue))
+            _logger.LogInformation($"user is not null and authenticated");
+
+            var optionClaim = user.Claims.FirstOrNone(x => x.Type == claimType);
+            _logger.LogInformation($"claim hasValue: <{optionClaim.HasValue}>");
+
+            var value = optionClaim.ValueOr(new Claim(claimType, string.Empty)).Value;
+            _logger.LogInformation($"value of claim: <{value}>");
+
+            if (string.IsNullOrWhiteSpace(value) || extractedQueryParameter.Equals(value))
```

Comment:
```suggestion
            if (extractedQueryParameter == value)
```
They are strings, so you can just do `==`

The suggestion is after separating the if into 2 different ones.